jstz_cli/src/utils.rs

```diff
@@ -0,0 +1,21 @@
+use std::fs;
+use std::io::{self, IsTerminal};
+
+pub fn from_file_or_id(data_or_file: String) -> String {
+    // try and read the file
+    fs::read_to_string(&data_or_file)
+        // file doesn't exist so assume it's raw data
+        .unwrap_or(data_or_file)
+}
+fn get_stdin() -> String {
+    let lines: Result<Vec<_>, _> = io::stdin().lines().collect();
+    lines.expect("Can't read from stdin").join("\n")
+}
+pub fn piped_input() -> Option<String> {
+    let stdin = io::stdin();
+    if !stdin.is_terminal() {
+        Some(get_stdin())
+    } else {
+        None
+    }
```

Comment:
Naive question: What is the behaviour of `is_terminal` in shell scripts?

Reply:
It's the same as in the terminal. It will return true if stdin is coming from the terminal (not from a pipe). The purpose of this line is to avoid the cli pausing and accepting input from the terminal. (Although we might want this behaviour)